test/server/core/gameSteps/abilityWindow/TriggeredAbilityWindow.spec.js

```diff
@@ -47,5 +47,61 @@ describe('Simultaneous triggers', function() {
                 expect(this.chewbacca.exhausted).toBe(false);
             });
         });
+
+        describe('Two units with a when defeated ability killing each other', function () {
```

Comment:
Also very cool to see that these work!! One thing to note though (might be worth a TODO for the benefit of anyone reading the tests) is that this case only works because the "when defeat" abilities are triggered by the defeat of the card that owns the ability.

For example, Iden's ability may not trigger correctly in this situation because if her defeat event resolves first, she will have left the field before the other unit defeat resolves.

I think defeat logic is definitely something for a different PR though just b/c the timing is so weird.